mem/mem/protocol.go

```diff
@@ -352,6 +352,10 @@ type ControlMsg struct {
 	DiscardTransations bool
 	Restart            bool
 	NotifyDone         bool
+	Drain              bool
```

Comment:
I get confused by these signals many times. We should consider adding some comments to clarify and help us remember. I rarely want comments, but I think this is the comment we should add.